jvb/src/main/kotlin/org/jitsi/videobridge/util/SafeExecutor.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.videobridge.util
+
+import org.jitsi.utils.logging2.LoggerImpl
+import org.json.simple.JSONObject
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Future
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.ThreadPoolExecutor
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.LongAdder
+
+/**
+ * Tracks the number of uncaught exceptions from tasks
+ */
+abstract class TaskServiceWrapper<T : ExecutorService>(
+    val name: String,
+    protected val delegate: T
+) {
+    private val logger = LoggerImpl(name)
+    private val numExceptions = LongAdder()
+
+    protected fun wrapTask(task: Runnable): Runnable {
+        return Runnable {
+            try {
+                task.run()
+            } catch (t: Throwable) {
+                logger.warn("Uncaught exception ($name): $t")
```

Comment:
I don't think we need `name` here, since it's already passed as the name of the logger.